File: lista java/at3.java
public class ContaBancaria {
   
    private String titular;
    private double saldo;

    
    public ContaBancaria(String titular, double saldoInicial) {
        this.titular = titular;
        this.saldo = saldoInicial;
    }

    
    public void depositar(double valor) {
        if (valor > 0) {
            saldo += valor;
            System.out.println("Depósito de R$" + valor + " realizado com sucesso.");
        } else {
            System.out.println("Valor inválido para depósito.");
        }
    }

    
    public void sacar(double valor) {
        if (valor > 0 && valor <= saldo) {
            saldo -= valor;
            System.out.println("Saque de R$" + valor + " realizado com sucesso.");
        } else {
            System.out.println("Saldo insuficiente ou valor inválido.");
        }
    }

    
    public double getSaldo() {
        return saldo;
    }

    public static void main(String[] args) {
        
        ContaBancaria conta = new ContaBancaria("João", 500);

        
        conta.depositar(200);
        conta.sacar(100);
        System.out.println("Saldo atual: R$" + conta.getSaldo());
        conta.sacar(700);  
    }
}

File: lista Go/at3.go
package main

import "fmt"


type ContaBancaria struct {
    Titular string
    saldo   float64  
}


func (c *ContaBancaria) Depositar(valor float64) {
    if valor > 0 {
        c.saldo += valor
        fmt.Printf("Depósito de R$%.2f realizado com sucesso.\n", valor)
    } else {
        fmt.Println("Valor inválido para depósito.")
    }
}


func (c *ContaBancaria) Sacar(valor float64) {
    if valor > 0 && valor <= c.saldo {
        c.saldo -= valor
        fmt.Printf("Saque de R$%.2f realizado com sucesso.\n", valor)
    } else {
        fmt.Println("Saldo insuficiente ou valor inválido.")
    }
}


func (c ContaBancaria) GetSaldo() float64 {
    return c.saldo
}

func main() {
    
    conta := ContaBancaria{Titular: "JOSE", saldo: 500}

   
    conta.Depositar(200)
    conta.Sacar(100)
    fmt.Printf("Saldo atual: R$%.2f\n", conta.GetSaldo())
    conta.Sacar(700)  
}
